hudi-utilities/src/main/java/org/apache/hudi/utilities/TableSizeStats.java

```diff
@@ -0,0 +1,411 @@
+      }
+
+      if (cfg.propsFilePath != null) {
+        List<String> filePaths = getFilePaths(cfg.propsFilePath, jsc.hadoopConfiguration());
+        for (String filePath : filePaths) {
+          logTableStats(filePath, startDate, endDate, cfg.partitionStats);
+        }
+      } else {
+        if (cfg.basePath == null) {
+          throw new HoodieIOException("Base path needs to be set.");
+        }
+        logTableStats(cfg.basePath, startDate, endDate, cfg.partitionStats);
+      }
+
+    } catch (Exception e) {
+      throw new HoodieException("Unable to do fetch table size stats." + cfg.basePath, e);
+    }
+  }
+
+  private List<String> getFilePaths(String propsPath, Configuration hadoopConf) {
+    List<String> filePaths = new ArrayList<>();
+    FileSystem fs = FSUtils.getFs(
+        propsPath,
+        Option.ofNullable(hadoopConf).orElseGet(Configuration::new)
+    );
+
+    try (BufferedReader reader = new BufferedReader(new InputStreamReader(fs.open(new Path(propsPath))))) {
+      String line = reader.readLine();
+      while (line != null) {
+        filePaths.add(line);
+        line = reader.readLine();
+      }
+    } catch (IOException ioe) {
+      LOG.error("Error reading in properties from dfs from file." + propsPath);
+      throw new HoodieIOException("Cannot read properties from dfs from file " + propsPath, ioe);
+    }
+    return filePaths;
+  }
+
+  private void logTableStats(String basePath, LocalDate startDate, LocalDate endDate, boolean partitionStats) throws IOException {
+
+    LOG.warn("Processing table " + basePath);
+    HoodieMetadataConfig metadataConfig = HoodieMetadataConfig.newBuilder()
+        .enable(false)
+        .build();
+    HoodieSparkEngineContext engineContext = new HoodieSparkEngineContext(jsc);
+    HoodieTableMetadata tableMetadata = HoodieTableMetadata.create(engineContext, metadataConfig, basePath,
+        FileSystemViewStorageConfig.SPILLABLE_DIR.defaultValue());
+    SerializableConfiguration serializableConfiguration = new SerializableConfiguration(jsc.hadoopConfiguration());
+
+    List<String> allPartitions = tableMetadata.getAllPartitionPaths();
+
+    final Histogram tableHistogram = new Histogram(new UniformReservoir(1_000_000));
+    allPartitions.forEach(partition -> {
+      // Partition name should conform to date format if startDate and/or endDate are specified. Otherwise, we don't
+      // need to parse partition name as date.
+      String dateString = partition;
+      if (partition.contains("=")) {
+        // Partition name may be in the form of "<column>=<date>". Try parsing out date.
+        String[] parts = partition.split("=");
+        dateString = parts[1].trim();
+      }
+
+      LocalDate partitionDate = null;
+      try {
+        if (startDate != null || endDate != null) {
+          partitionDate = LocalDate.parse(dateString, DATE_FORMATTER);
+        }
+        LOG.info("Parsed " + partition);
+      } catch (DateTimeParseException dtpe) {
```

Comment:
can we move this parsing outside of map or for each call.

Reply:
Added a check outside the forEach call to catch any date format mismatch early.